README.md

```diff
@@ -139,60 +150,81 @@ chain.set({
   foo: 1
 , 'a.b': '#foo'
 , bar: '!random'
+, combine: '!template:"foo-{{#foo}}, ab-{{#a.b}}"'
 })
 await chain.execute()
 
-> {foo: 1, bar: 'af4b31', a: {b: 1}}
+> {foo: 1, bar: 'af4b31', a: {b: 1}, combine: 'foo-1, ab-1'}
 ```
 
-## `lookup(<String>|<Object>|<Array>)`
-
-### Parameters
-
-* **input** `Any`: The string path or object to lookup
-
-Resolves lookup value. This function is generally used by actions to resolve values
-from the chain as it is executes.
-
-* If the input is a string, and starts with a `#` symbol it is considered to be a lookup path and will
-  return the requested value from internal state if found.
-* If the input is a string, and starts with a `!` symbol it is considered a function call and will call
-  any registered actions from the chain instance and store the result in internal state.
-* If the input is an object, each key in the object will be resolved following the above rules
-* If the input is an array, each item in the array will be resolved following the above rules
+## `map(collection, iterator, label)`
 
-## `map(<Array>|<String>, <Function>, <String>)`
+Takes a collection (array, or `#lookup` value), and applies an iterator function to each item.
+The return value from the iterator will be the final result.  Results are stored in an array,
+keyed by `label`, and order is maintained. This is a chain action resolved by [execute](#execute).
```

Comment:
maybe: "Results are stored in an array and placed into the chain state at `label` ..." 

actually, maybe it's worth calling out in/near the "Program Flow" section the way that results get persisted into the chain, so you don't have to mention it in the built-ins? It's alluded to in the examples, but it might deserve some explicitness.

Free sample:

### State

The result returned from each action on the chain is stored in the `state` object. The `label` parameter of task signature will be used as the property for the return value in `state` . If omitted, the action name will be used instead.

```js
const state = await chain
  .foo('first')
  .foo('second', 'foo_two')
  .bar({baz: 'foobarbaz'}, 'bar_label')
  .execute()

Object.keys(state) // ['foo', 'foo_two', 'baz_label']
```

Reply:
This was definitely an area for improvement.  I like your suggestion.  There's a few things that are just casually mentioned when they deserve a callout.